### PR TITLE
DOCS: add heap checksum information to gpstart.

### PR DESCRIPTION
**Pull request by mkiyama**
Added
--new option --skip-heap-checksum-validation
--updated -m option. no checksum validation.
--New information about checking heap checksum validation

### CODE REVIEW

**Comment by Chibin** [comment]
This is a follow up PR based on https://github.com/greenplum-db/gpdb/pull/2921

File: gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml [hunk 1]
```diff
@@ -30,8 +31,14 @@
                 efficiently. If new hosts are added to the system, you must manually remove this
                 file from the <codeph>gpadmin</codeph> user's home directory. The utility will
                 create a new hosts cache file at the next startup.</p>
+            <p>As part of the startup process, the utility checks the consistency of heap checksum
+                setting among the Greenplum Database master and segment instances, either enabled or
+                disabled on all instances. If the heap checksum setting is different among the
+                instances, an error is returned and Greenplum Database does not start. The
+                validation can be disabled by specifying the option
```
Comment (mkiyama) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml]: Is this clearer? 

To try to fix checksum consistency errors, you can specify the option --skip-heap-checksum-validation to disable the check that occurs during start up.

Comment (dyozie) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml]: Actually, the original text is probably more accurate.  My issue after the first reading was that gpstart might be checking whether they incorrectly configured checksums (ie. enabled/disabled checksums differently on different hosts, or configured `ignore_checksum_failure` differently on different hosts).  But really those two things are system-wide configurations.   So the gpstart check is for some anomaly where a segment somehow has checksums configured differently.  Maybe there needs to be a note about how that situation could occur, and if it does how would a reader go about fixing it.

File: gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml [hunk 2]
```diff
@@ -76,6 +85,16 @@
...
                 </plentry>
                 <plentry>
+                    <pt>--skip-heap-checksum-validation</pt>
+                    <pd>During startup, the utility does not validate the consistency of the heap
+                        checksum setting among the Greenplum Database master and segment instances.
+                        The default is to ensure that the heap checksum setting is the same on all
+                        instances, either enabled or disabled.<note type="warning">Starting
+                            Greenplum Database without this validation could lead to data loss. This
+                            option should only be used to start Greenplum Database to perform
+                            maintenance or recovery tasks.</note></pd>
```
Comment (asimrp) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml]: This is a one-way path.  If Greenplum is started with data checksums inconsistently set across the cluster, there is no way to get back data protection with checksums without reinitializing the cluster (gpinitsystem).

**Comment by mkiyama** [comment]
@asimrp @dyozie 
Addressed your review comments by adding more information to the end of the overview section.
I think it works.

File: gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml [hunk 3]
```diff
@@ -36,9 +36,16 @@
...
-                system using <codeph>gpinitsystem</codeph>.</p>
+                    <codeph>--skip-heap-checksum-validation</codeph>. For more information about
+                heap checksums, see <xref
+                    href="../../admin_guide/highavail/topics/g-overview-of-high-availability-in-greenplum-database.xml"
+                    format="dita" scope="peer">Overview of Greenplum Database High
+                    Availability</xref> in the <cite>Greenplum Database Admininstartor
+                Guide</cite>.</p>
+            <note>Before you can start a Greenplum Database system, you must have initialized the
+                system using <codeph>gpinitsystem</codeph>. Enabling or disabing heap checksums is
+                performed system intitalization with and cannot be changed after
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml]: "system intitalization with" -> "with system intitalization"

File: gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml [hunk 4]
```diff
@@ -76,6 +92,16 @@
...
                 </plentry>
                 <plentry>
+                    <pt>--skip-heap-checksum-validation</pt>
+                    <pd>During startup, the utility does not validate the consistency of the heap
+                        checksum setting among the Greenplum Database master and segment instances.
+                        The default is to ensure that the heap checksum setting is the same on all
+                        instances, either enabled or disabled. <note type="warning">Starting
+                            Greenplum Database without this validation could lead to data loss. This
+                            option should only be used to start Greenplum Database to perform
+                            maintenance or recovery tasks.</note></pd>
```
Comment (asimrp) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpstart.xml]: The last sentence is diluting the gravity of the situation, at least when I read it.  The previous sentence is correctly conveying the problem, though.  So I'm fine here.

**Pull request merged**